### PR TITLE
Changing Dockerfile to use alpine:3.6 as a basis

### PR DESCRIPTION
**Pull request by marians**
Moving from Ubuntu to alpine:3.6, as it's both smaller and more secure.

image

If the `curl` and `wget` binaries are only needed for the build, we can also make sure to get rid of them, to make the image smaller. Same of course for `bash` and `git`, which I have only added so the plugin's `cnr.sh` script can be executed. Thanks for any hints!

Update:

image

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
We don't actually need `wget` and `curl` - using `curl -o` should suffice to replace `wget`.

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-
-RUN apt-get -y update \
-    && apt-get -y install \
-    curl wget
+FROM alpine:3.6
 
+ENV ALPINE_GLIBC_VERSION 2.26-r0
 ENV HELM_VERSION 2.6.2
 
+RUN apk update && apk --no-cache add ca-certificates openssl curl wget git bash \
```
Comment (JosephSalisbury) [inline Dockerfile]: We probably don't actually need wget and curl - using curl -o should suffice to replace wget.

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
     && mv ./linux-amd64/helm /bin/helm \
     && rm -rf ./linux-amd64
 
-RUN mkdir -p ~/.helm/plugins/ \
-    && wget https://github.com/app-registry/appr-helm-plugin/releases/download/v0.5.1/registry-helm-plugin.tar.gz -qO- | tar xzf - registry \
-    && mv ./registry ~/.helm/plugins/ \
-    && ~/.helm/plugins/registry/cnr.sh upgrade-plugin
+RUN mkdir -p ~/.helm/plugins \
+    && cd ~/.helm/plugins \
+    && git clone https://github.com/app-registry/appr-helm-plugin.git registry \
```
Comment (JosephSalisbury) [inline Dockerfile]: Can we `curl` this, instead of using `git`? If you wanna get the image size down.

File: Dockerfile [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
-RUN apt-get -y update \
-    && apt-get -y install \
-    curl wget
+FROM alpine:3.6
 
+ENV ALPINE_GLIBC_VERSION 2.26-r0
 ENV HELM_VERSION 2.6.2
 
+RUN apk update && apk --no-cache add ca-certificates openssl curl wget git bash \
+    && wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://raw.githubusercontent.com/sgerrand/alpine-pkg-glibc/master/sgerrand.rsa.pub \
```
Comment (JosephSalisbury) [inline Dockerfile]: Why do we need to update glibc?

Comment (marians) [reply Dockerfile]: alpine doesn't provide glibc at all. Here we install it, assuming that the `draughtsman` binary will need it to run.

File: Dockerfile (hunk 2 above)
Comment (marians) [reply Dockerfile]: Yeah, I might get back to the way that part worked before. `git` is what's recommended in the appr plugin docs. However, with cloning the repo's master you never know what you get.

File: Dockerfile (hunk 3 above)
Comment (JosephSalisbury) [reply Dockerfile]: Does it? Do statically compiled binaries still require `glibc`?

File: Dockerfile (hunk 1 above)
Comment (marians) [reply Dockerfile]: Replaced wget with curl calls

File: Dockerfile (hunk 2 above)
Comment (marians) [reply Dockerfile]: Done

**Comment by marians** [comment]
@JosephSalisbury please check again

**Comment by JosephSalisbury** [comment]
Still curious about the glibc thing:
> assuming that the `draughtsman` binary will need it to run.

do we need it?

**Comment by marians** [comment]
Seems to work without glibc. Good call!

File: .circleci/config.yml [hunk 4]
```diff
@@ -16,6 +16,10 @@ jobs:
           name: architect build
           command: ./architect build
 
+      - run:
```
Comment (JosephSalisbury) [inline .circleci/config.yml]: This isn't necessary. `architect` runs `--help` already

**Comment by marians** [comment]
Thanks Joe!

**Pull request merged**